cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/distributed/AppFabricServiceManager.java

```diff
@@ -51,10 +65,22 @@ public int getMaxInstances() {
   public SystemServiceLiveInfo getLiveInfo() {
     SystemServiceLiveInfo.Builder builder = SystemServiceLiveInfo.builder();
 
-    Containers.ContainerInfo containerInfo = new Containers.ContainerInfo(Containers.ContainerType.SYSTEM_SERVICE,
-                                                                          Constants.Service.APP_FABRIC_HTTP, null, null,
-                                                                          hostname.getHostName(), null, null, null);
-    builder.addContainer(containerInfo);
+    try {
+      SortedMap<Integer, LeaderElectionInfoService.Participant> participants =
+        electionInfoService.getParticipants(ELECTION_PARTICIPANTS_TIMEOUT_MS, TimeUnit.MILLISECONDS);
+
+      for (LeaderElectionInfoService.Participant participant : participants.values()) {
+        builder.addContainer(new Containers.ContainerInfo(Containers.ContainerType.SYSTEM_SERVICE,
+                                                          Constants.Service.APP_FABRIC_HTTP, null, null,
+                                                          participant.getHostname(), null, null, null));
+      }
+    } catch (Exception e) {
```

Comment:
It happens when there is interrupt exception or timeout in getting the map (the chance is super small). In either case, I think we can safely ignore without logging.